### ISSUE TITLE
[Feature request] Customize note title shown in tree

### ISSUE THREAD

**Issue by wwwald** [zadam/trilium#1269]
Using the latest Trilium 0.44.4.

All my journal notes have a label "project", containing the project name they belong to(for instance, "my-website", "my-next-album"). It would be nice if the Tree view could also show this label, along with the note title.

In a perfect world, I would be able to dynamically customize the note title shown in the treeview, to also show this "project" label value. For instance, the tree title could be made configurable via a "tree title template" setting.

Assume that the current Trilium release shows this tree view of my notes for a given day:

![image](https://user-images.githubusercontent.com/39531/94272557-8061b480-ff43-11ea-9d30-2ea9f3f5f49a.png)

With this new feature, I would customize the "tree title template" as follows:

    [note.labels.project] note.title

This would turn the view above into:

![image](https://user-images.githubusercontent.com/39531/94272660-9ec7b000-ff43-11ea-8c93-83d06687892e.png)

I'm not yet sure at what level such a "tree title template" should be defined - I can imagine that it would not apply to the whole wiki tree at once, but rather be defined at the subtree level.

Any value in this?

**Comment by zadam** [comment]
Hi, I understand your use case and this would be feasible to implement. My biggest concern is how widely useful this would be.

**Comment by wwwald** [comment]
Fair enough - I can definitely see that this may not be of interest to a lot of users.  
I'm not going to nag you for it :-)

**Comment by dstengle** [comment]
I would love to have this and have it be by tree. I'd like to have the date prefixed for journalized type entries and then maybe a project for other types of entries.

It would be nice to customize the note title at creation time with the template to just insert the date by default.

**Comment by s-lt** [comment]
I would love such a feature as well; FWIW, one of my most beloved features of any software is the way you are displaying things in [Foobar2k](https://wiki.hydrogenaud.io/index.php?title=Foobar2000:Title_Formatting_Reference#:~:text=A%20title%20formatting%20script%20consists,%2F%2F%20this%20is%20a%20comment%20.).
There, you are basically using a scripting language, which draws its input from the tags of the music files to be displayed.

For Trilium, this would be nice as a inheritable attribute, where you can substitute a note's title with something like "#todo : #date - #title - #involvedpeople"
Of course, getting things like $if and $rgb and other "proper" functions and formatting stuff would be also great, but I got no clue how feasible any of this is.
